Source/SwiftyJSON.swift

```diff
@@ -123,6 +123,42 @@ public struct JSON {
         }
         self.init(dictionary)
     }
+    
+    /**
+     Merges another JSON into this JSON, whereas primitive values which are not present in this JSON are getting added, 
+     present values getting overwritten, array values getting appended and nested JSONs getting merged the same way.
+ 
+     - parameter other: The JSON which gets merged into this JSON
+     */
+    public mutating func merge(_ other: JSON) {
+        if self.type == other.type {
+            switch self.type {
+            case .dictionary:
+                for (key, _) in other {
+                    self[key].merge(other[key])
+                }
+            case .array:
+                self = JSON(self.arrayValue + other.arrayValue)
+            default:
+                self = other
+            }
+        } else {
+            self = other
```

Comment:
Can you clarify what you mean by

 > I wouldn't necessary treat that as an error, since the main identifier in a JSON is the `key`

I think SwiftyJSON's goal is to be as safe as possible, which is why there is no force unwrap, that all getters are optional and that non-optional `*Value` have fallback values. My point is that we should always make sure that errors are propagated back to the client and that we corrupt data without one of those errors.

If someone did want to replace the entire JSON when merging with conflicting types, they could simply do:

```
let json: JSON
do {
    json = try json.merged(with: otherJSON)
} catch {
    json = otherJSON
}
```

How does that sound?